judge/judgedaemon.main.php

```diff
@@ -46,6 +46,28 @@ function read_credentials()
     }
 }
 
+
+function setup_curl_handle()
+{
+    global $curl_handle, $restuser, $restpass;
+
+    if ( empty($curl_handle) ) {
+        $curl_handle = curl_init();
+        curl_setopt($curl_handle, CURLOPT_USERAGENT, "DOMjudge/" . DOMJUDGE_VERSION);
+        curl_setopt($curl_handle, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
+        curl_setopt($curl_handle, CURLOPT_USERPWD, $restuser . ":" . $restpass);
+        curl_setopt($curl_handle, CURLOPT_RETURNTRANSFER, true);
+    }
+}
+
+function close_curl_handle()
+{
+    global $curl_handle;
+    if ( ! empty($curl_handle) ) {
+        curl_close($curl_handle);
+    }
```

Comment:
currently it's not necessary from the calls that I can see, but we should reset the handle here

Reply:
agreed

Reply:
Agreed, `curl_close()` promises to "delete the handle" but that's not fully resetting `$curl_handle` apparently.